src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java

```diff
@@ -32,98 +28,111 @@
         return createTestTable("integer");
     }
 
     @Override
-    protected VirtualSchemaTestTable<Double> createDoubleVirtualSchemaTable() throws SQLException {
+    protected SingleValueVirtualSchemaTestTable<Double> createDoubleVirtualSchemaTable() {
         return createTestTable("real");
     }
 
-    private <T> VirtualSchemaTestTable<T> createTestTable(final String type) throws SQLException {
-        final Table table = postgresSchema.createTableBuilder(POSTGRES_TABLE_NAME)//
-                .column("my_column", type).build();
-        this.createdTables.add(table);
-        refreshVirtualSchema();
-        final String tableName = virtualSchema.getFullyQualifiedName() + "." + POSTGRES_TABLE_NAME;
-        return new VirtualSchemaTestTable<>(tableName, getTruncateValueSetter(table));
+    @Override
+    protected SingleValueVirtualSchemaTestTable<Boolean> createBooleanVirtualSchemaTable() {
+        return createTestTable("boolean");
     }
 
-    private <T> VirtualSchemaTestTable.SingleRowTableProvisioner<T> getTruncateValueSetter(final Table table) {
-        return (value) -> {
-            truncateTable(table);
-            table.insert(value);
+    private <T> SingleValueVirtualSchemaTestTable<T> createTestTable(final String type) {
+        return new SingleValuePostgresVirtualSchemaTestTable<>() {
+            @Override
+            protected Table createTable(final Schema schema) {
+                return schema.createTableBuilder(getUniqueIdentifier())//
+                        .column("my_column", type).build();
+            }
         };
     }
 
-    @Override
-    protected String createBooleanVirtualSchemaTableWithSingleTrueValue() throws SQLException {
-        final Table table = postgresSchema.createTableBuilder(POSTGRES_TABLE_NAME)//
-                .column("boolean", "boolean").build()//
-                .insert(true);
-        this.createdTables.add(table);
-        refreshVirtualSchema();
-        return virtualSchema.getFullyQualifiedName() + "." + POSTGRES_TABLE_NAME;
+    protected Connection createExasolConnection() throws SQLException {
+        return SETUP.getExasolContainer().createConnection();
     }
 
     @Override
-    protected VirtualSchemaTestTable<String> createStringVirtualSchemaTable() throws SQLException {
+    protected SingleValueVirtualSchemaTestTable<String> createStringVirtualSchemaTable() {
         return createTestTable("VARCHAR(500)");
     }
 
-    private void truncateTable(final Table table) throws SQLException {
-        SETUP.getPostgresqlStatement().executeUpdate("TRUNCATE TABLE " + table.getFullyQualifiedName());
-    }
-
-    @Override
-    protected void deleteCreatedTables() {
-        this.createdTables.forEach(Table::drop);
-        this.createdTables.clear();
+    private static abstract class PostgresVirtualSchemaTestTable implements VirtualSchemaTestTable {
+        private final VirtualSchema virtualSchema;
+        private final Table table;
```

Comment:
looks to me that
`private final VirtualSchema virtualSchema;
private final Table table;`
belong to a super class.

Reply:
no, Table is test-db-builder which is implementation detail

Reply:
are other implementations, eg, for MySQL, SQLServer, etc, gonna use a different approach here, or you think this might change among dialects?

Reply:
yes. For example elasticsearch, or other no-sql dialects are different

Reply:
until now test-db-builder only supportst postgres, mysql and exasol